src/app/components/rawdata/rawdata.component.ts

```diff
@@ -25,44 +29,57 @@ export class RawDataComponent implements OnInit, OnDestroy {
   private _counter = 1;
 
   /** Constructor */
-  constructor(private _stompService: STOMPService,
-    private _configService: ConfigService) { }
+  constructor(private _stompService: STOMPService) { }
 
   ngOnInit() {
-    // Get configuration from config service...
-    this._configService.getConfig('api/config.json').then(
-      config => {
-        // ... then pass it to (and connect) STOMP:
-        this._stompService.configure(config);
-        this._stompService.try_connect().then(this.on_connect);
-      }
-    );
+    this.subscribed = false;
+
+    // Store local reference to Observable
+    // for use with template ( | async )
+    this.subscribe();
+  }
+
+  public subscribe() {
+    if(this.subscribed) {
+      return;
+    }
+
+    // Stream of messages
+    this.messages = this._stompService.subscribe("/topic/ng-demo-sub");
+
+    // Subscribe a function to be run on_next message
+    this.subscription = this.messages.subscribe(this.on_next);
+
+    this.subscribed = true;
+  }
+
+  public unsubscribe() {
+    if(!this.subscribed) {
+      return;
+    }
+
+    // This will internally unsubscribe from Stomp Broker
+    // There are two subscriptions - one created explicitly, the other created in the template by use of 'async'
+    this.subscription.unsubscribe();
+    this.subscription = null;
+    this.messages = null;
+
+    this.subscribed = false;
   }
 
   ngOnDestroy() {
-    this._stompService.disconnect();
+    this.unsubscribe();
   }
 
-  public onClick() {
+  public onSendMessage() {
     const _getRandomInt = (min, max) => {
       return Math.floor(Math.random() * (max - min + 1)) + min;
     };
-    this._stompService.publish(`{ type: "Test Message", data: [ ${this._counter}, ${_getRandomInt(1, 100)}, ${_getRandomInt(1, 100)}] }`);
+    this._stompService.publish("/topic/ng-demo-sub", `{ type: "Test Message", data: [ ${this._counter}, ${_getRandomInt(1, 100)}, ${_getRandomInt(1, 100)}] }`);
```

Comment:
Hard-coded topic for demo: Is there a rationale for this, or should this be parameterized (and/or in `config.json`?)